modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -56,7 +55,13 @@ sealed class IO<out A> : IOOf<A> {
             IO { callback(Left(t)) }
           }
 
-          IORunLoop.startCancelable(fa, conn2, mapUnit)
+          //DEV: If fa cancels conn2 like so `conn.cancel().map { cb(Right(Unit)) }`
+          //It doesn't run the stack of conn2, instead the result is seen it cb of startCancelable.
+          IORunLoop.startCancelable(fa, conn2) {
+
+            if (it.fold({ e -> e == OnCancel.CancellationException }, { false }) && conn.isNotCanceled()) IORunLoop.start(conn.cancel(), mapUnit)
```

Comment:
nit: take the fold into boolean in a new variable. Put the comment on that variable instead